src/main/java/gregtech/common/metatileentities/multi/electric/MetaTileEntityProcessingArray.java

```diff
@@ -0,0 +1,303 @@
+                .where('#', isAirPredicate()).build();
+    }
+
+    public IBlockState getCasingState() {
+        return MetaBlocks.METAL_CASING.getState(BlockMetalCasing.MetalCasingType.TUNGSTENSTEEL_ROBUST);
+    }
+
+    @Override
+    public ICubeRenderer getBaseTexture(IMultiblockPart sourcePart) {
+        return Textures.ROBUST_TUNGSTENSTEEL_CASING;
+    }
+
+    @Nonnull
+    @Override
+    protected OrientedOverlayRenderer getFrontOverlay() {
+        return Textures.PROCESSING_ARRAY_OVERLAY;
+    }
+
+    @Override
+    public boolean canBeDistinct() {
+        return true;
+    }
+
+    @Override
+    public boolean hasMaintenanceMechanics() {
+        return true;
+    }
+
+    protected class ProcessingArrayWorkable extends MultiblockRecipeLogic {
+
+        ItemStack currentMachineStack = null;
+        ItemStack oldMachineStack = null;
+        //The Voltage Tier of the machines the PA is operating upon, from GTValues.V
+        private int machineTier;
+        //The maximum Voltage of the machines the PA is operating upon
+        private long machineVoltage;
+        //The Recipe Map of the machines the PA is operating upon
+        private RecipeMap<?> activeRecipeMap;
+
+        public ProcessingArrayWorkable(RecipeMapMultiblockController tileEntity) {
+            super(tileEntity);
+        }
+
+        @Override
+        public void invalidate() {
+            super.invalidate();
+            // Reset locally cached variables upon invalidation
+            currentMachineStack = null;
+            oldMachineStack = null;
+            machineTier = 0;
+            machineVoltage = 0L;
+            activeRecipeMap = null;
+        }
+
+        /**
+         * Checks if a provided Recipe Map is valid to be used in the processing array
+         * Will filter out anything in the config blacklist, and also any non-singleblock machines
+         * @param recipeMap The recipeMap to check
+         * @return {@code true} if the provided recipeMap is valid for use
+         */
+        @Override
+        public boolean isRecipeMapValid(RecipeMap<?> recipeMap) {
+            if (findMachineInBlacklist(recipeMap.getUnlocalizedName())) {
+                return false;
+            }
+
+            RecipeBuilder<?> recipeBuilder = recipeMap.recipeBuilder();
+
+
+            return (recipeBuilder instanceof SimpleRecipeBuilder ||
+                    recipeBuilder instanceof IntCircuitRecipeBuilder ||
+                    recipeBuilder instanceof UniversalDistillationRecipeBuilder ||
+                    recipeBuilder instanceof CircuitAssemblerRecipeBuilder) &&
+                    //These are multiblocks that are also SimpleRecipeBuilder. Need a better way to exclude these
+                    !(recipeMap == RecipeMaps.VACUUM_RECIPES || recipeMap == RecipeMaps.LARGE_CHEMICAL_RECIPES ||
+                            recipeMap == RecipeMaps.ASSEMBLY_LINE_RECIPES || recipeMap == RecipeMaps.PROCESSING_ARRAY_RECIPES);
```

Comment:
All that should be needed is a check if the current machine is an instance of `WorkableTieredMetaTileEntity`, and if the recipemap of that machine is in the blacklist.